eclair-core/src/main/scala/fr/acinq/eclair/blockchain/electrum/ElectrumClient.scala

```diff
@@ -93,10 +96,12 @@ class ElectrumClient(serverAddress: InetSocketAddress, ssl: SSL)(implicit val ec
       ch.pipeline.addLast(new JsonRPCRequestEncoder)
       // error handler
       ch.pipeline.addLast(new ExceptionHandler)
+      socksProxy_opt.foreach(params => ch.pipeline().addFirst(new Socks5ProxyHandler(params.address)))
     }
   })
 
   // Start the client.
+  socksProxy_opt.foreach(params => b.resolver(NoopAddressResolverGroup.INSTANCE))
```

Comment:
Can you explain that? Maybe a comment? Browsing to it in code doesn't reveal anything very useful...